src/commands/upload_dart_symbol_map.rs

```diff
@@ -0,0 +1,167 @@
+use std::borrow::Cow;
+use std::ffi::OsStr;
+use std::fmt::{Display, Formatter, Result as FmtResult};
+use std::path::Path;
+
+use anyhow::{bail, Context as _, Result};
+use clap::{Arg, ArgMatches, Command};
+
+use crate::api::{Api, ChunkUploadCapability};
+use crate::config::Config;
+use crate::constants::{DEFAULT_MAX_DIF_SIZE, DEFAULT_MAX_WAIT};
+use crate::utils::args::ArgExt as _;
+use crate::utils::chunks::{upload_chunked_objects, Assemblable, ChunkOptions, Chunked};
+use crate::utils::dif::DifFile;
+use symbolic::common::DebugId;
+
+struct DartSymbolMapObject {
+    bytes: Vec<u8>,
+    name: String,
+    debug_id: DebugId,
+}
```

Comment:
This allows the struct to store references rather than need to own data. For the `bytes`, this is particularly important, as we avoid yet another in-memory copy of the entire mapping file

```suggestion
struct DartSymbolMapObject<'a> {
    bytes: &'a [u8],
    name: &'a str,
    debug_id: DebugId,
}
```